src/liballoc/tests/vec.rs

```diff
@@ -726,6 +765,31 @@ fn test_into_iter_clone() {
     assert_eq!(it.next(), None);
 }
 
+#[test]
+fn test_into_iter_leak() {
+    static mut DROPS: i32 = 0;
+
+    struct D(bool);
+
+    impl Drop for D {
+        fn drop(&mut self) {
+            unsafe {
+                DROPS += 1;
+            }
+
+            if self.0 {
+                panic!("panic in `drop`");
+            }
+        }
+    }
+
+    let v = vec![D(false), D(true), D(false)];
```

Comment:
There's another memory leak in this test: the `Vec` backing store does not get deallocated. See https://github.com/rust-lang/rust/issues/69770.